Fixes #507 in LTC - vulnerability in der_decode_utf8_string

jbech-linaro opened the pull request:
Fix a vulnerability in der_decode_utf8_string as specified here:
https://github.com/libtom/libtomcrypt/issues/507

Patch manually picked from:
  https://github.com/libtom/libtomcrypt/commit/25c26a3b7a9ad8192ccc923e15cf62bf0108ef94

[Joakim Bech: Extended commmit message]
Signed-off-by: Luigi Coniglio <werew@ret2libc.com>
Acked-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (QEMU v7)

<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

jenswi-linaro commented:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

jbech-linaro commented:
Squashed, rebased and tag(s) applied (should be) ready for merge! Thanks!

jforissier commented:
Minor comments...
1. The author is != first S-o-b
2. Since you modified the patch (message) I tend to think you should give your S-o-b too. And the `[]` comment normally appears just before that `S-o-b`, like so:
```
Signed-off-by: Luigi Coniglio <werew@ret2libc.com>
[Joakim Bech: Extended commmit message]
Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Acked-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (QEMU v7)
```

3. s/commmit/commit/
4. Function names are normally cited with parentheses: der_decode_utf8_string()
5. `Acked-by: Jerome Forissier <jerome@forissier.org>`

jbech-linaro commented:
> 1. The author is != first S-o-b
> 2. Since you modified the patch (message) I tend to think you should give your S-o-b too. And the `[]` comment normally appears just before that `S-o-b`, like so:
> 
> ```
> Signed-off-by: Luigi Coniglio <werew@ret2libc.com>
> [Joakim Bech: Extended commmit message]
> Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
> Acked-by: Joakim Bech <joakim.bech@linaro.org>
> Tested-by: Joakim Bech <joakim.bech@linaro.org> (QEMU v7)
> ```

All fixed

> 
> 1. s/commmit/commit/

Good catch, fixed!

> 2. Function names are normally cited with parentheses: der_decode_utf8_string()

Indeed, just limited amount of space for subject, so I try to keep it short. But ... added ...

> 3. `Acked-by: Jerome Forissier <jerome@forissier.org>`

Added.

Now ... maybe ... 
Squashed, rebased and tag(s) applied (should be) ready for merge! Thanks!

jforissier commented:
```
$ git show FETCH_HEAD
commit 20454ffbcf3dc504dd9338a4930436c704be9790
Author: werew <werew@ret2libc.com>
[...]    
    Signed-off-by: Luigi Coniglio <werew@ret2libc.com>
[...]
```

jbech-linaro commented:
> $ git show FETCH_HEAD

Then we try again. I feel a bit uneasy to do this many changes to a patch (message) without being the author. It looks nothing at all like the [original commit](https://github.com/libtom/libtomcrypt/commit/25c26a3b7a9ad8192ccc923e15cf62bf0108ef94) any longer.

jbech-linaro commented:
New patch pushed if it wasn't clear with the previous message.
